experimental/traffic-portal/src/app/core/users/roles/detail/role-detail.component.ts

```diff
@@ -0,0 +1,110 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { Location } from "@angular/common";
+import { Component, OnInit } from "@angular/core";
+import { MatDialog } from "@angular/material/dialog";
+import { ActivatedRoute } from "@angular/router";
+import { ResponseRole } from "trafficops-types";
+
+import { UserService } from "src/app/api";
+import { DecisionDialogComponent } from "src/app/shared/dialogs/decision-dialog/decision-dialog.component";
+import { NavigationService } from "src/app/shared/navigation/navigation.service";
+
+/**
+ * AsnDetailComponent is the controller for the ASN add/edit form.
+ */
+@Component({
+	selector: "tp-role-detail",
+	styleUrls: ["./role-detail.component.scss"],
+	templateUrl: "./role-detail.component.html"
+})
+export class RoleDetailComponent implements OnInit {
+	public new = false;
+	public permissions = "";
+	public role!: ResponseRole;
+	constructor(private readonly route: ActivatedRoute, private readonly userService: UserService,
+		private readonly location: Location, private readonly dialog: MatDialog,
+		private readonly header: NavigationService) {
+	}
+
+	/**
+	 * Angular lifecycle hook where data is initialized.
+	 */
+	public async ngOnInit(): Promise<void> {
+		const role = this.route.snapshot.paramMap.get("name");
+		if (role === null) {
+			console.error("missing required route parameter 'name'");
+			return;
+		}
+
+		if (role === "new") {
+			this.header.headerTitle.next("New Role");
+			this.new = true;
+			this.role = {
+				description: "",
+				name: "",
+				permissions: []
+			};
+			return;
+		}
+
+		this.role = await this.userService.getRoles(role);
+		this.permissions = this.role.permissions?.join("\n")??"";
+		this.header.headerTitle.next(`Role: ${this.role.name}`);
+	}
+
+	/**
+	 * Deletes the current ASN.
+	 */
+	public async deleteRole(): Promise<void> {
+		if (this.new) {
+			console.error("Unable to delete new role");
+			return;
+		}
+		const ref = this.dialog.open(DecisionDialogComponent, {
+			data: {message: `Are you sure you want to delete role ${this.role.name} with description ${this.role.description}`,
+				title: "Confirm Delete"}
+		});
+		ref.afterClosed().subscribe(result => {
+			if(result) {
+				this.userService.deleteRole(this.role);
+				this.location.back();
+			}
+		});
+	}
+
+	/**
+	 * Updates permissions list from a string to an array.
+	 */
+	public async updatePermissions(): Promise<void> {
+		this.role.permissions = this.permissions.split("\n");
+	}
+
+	/**
+	 * Submits new/updated ASN.
+	 *
+	 * @param e HTML form submission event.
+	 */
+	public async submit(e: Event): Promise<void> {
+		e.preventDefault();
+		e.stopPropagation();
+		if(this.new) {
+			this.role = await this.userService.createRole(this.role);
+			this.new = false;
+		} else {
+			this.role = await this.userService.updateRole(this.role);
+		}
```

Comment:
When you're done either creating or updating the role, you should use the router and the navigation service to change the URL (without adding to history) and update the page title, because the Role's name may have changed (or been decided, if it was created).

Reply:
Page title update is broken on all pages: https://github.com/apache/trafficcontrol/issues/7418